services/api-server/src/simcore_service_api_server/_service_programs.py

```diff
@@ -0,0 +1,35 @@
+from typing import Annotated
+
+from fastapi import Depends
+from models_library.basic_types import VersionStr
+from models_library.services_enums import ServiceType
+
+from .models.schemas.programs import Program, ProgramKeyId
+from .services_rpc.catalog import CatalogService
+
+
+class ProgramService:
+    _catalog_service: CatalogService
+
+    def __init__(
+        self, _catalog_service: Annotated[CatalogService, Depends(CatalogService)]
```

Comment:
uaua i really like this approach of annotating with a dependency in the constructor.
The constructor can be tested e.g. with a mock catalog_service (e.g. with a mock) since  and at the same time used in the controller where the dependency injection will conect with the apps;s initialized CatalogServcice

Very nice!